### PR TITLE
Improvement/diagrams focus movement

### LINKED ISSUE #248: Multi-diagrams page: diagrams focus movement
## Description

On the Multi-diagrams page, in order to zoom in/out, we want to move the diagrams container to focus the screen on specific area.

### PR DESCRIPTION
**Pull request by k-y0u**
resolves #248

### CODE REVIEW

File: src/pages/DiagramsPage.vue [hunk 1]
```diff
@@ -78,6 +90,18 @@ function zoom(plus) {
...
+/**
+ * Pan on diagrams container according on the delta.
+ * @param {*} delta - Delta of distance (in pixels) since handler was called last time.
+ */
+function handlePan({ delta }) {
+  translate.x += delta.x;
+  translate.y += delta.y;
+
+  const div = document.getElementById('diagrams-container');
+  div.style.translate = `${translate.x}px ${translate.y}px`;
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:102]: no jump line after `const ...` ?

File: src/pages/DiagramsPage.vue [hunk 2]
```diff
@@ -78,6 +90,18 @@ function zoom(plus) {
   div.style.scale = scale.value;
 }
 
+/**
+ * Pan on diagrams container according on the delta.
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:94]: Not sure if this is understandable

File: cypress/e2e/DiagramsPage/focus.movement.diagrams.feature [hunk 3]
```diff
@@ -0,0 +1,38 @@
+Feature: Test diagrams page: move all diagrams
+  Scenario: Should move all diagrams
```
Comment (justinehell) [inline cypress/e2e/DiagramsPage/focus.movement.diagrams.feature:2]: Jump line before

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/pages/DiagramsPage.vue [hunk 4]
```diff
@@ -1,5 +1,9 @@
 <template>
-  <q-page>
+  <q-page
+    id="diagrams-page"
+    v-touch-pan.mouse="handlePan"
+    :data-cy="`diagrams_page`"
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: done

File: src/pages/DiagramsPage.vue (hunk 1 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: done

File: src/pages/DiagramsPage.vue (hunk 2 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue:94]: done

File: cypress/e2e/DiagramsPage/focus.movement.diagrams.feature (hunk 3 above)
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/focus.movement.diagrams.feature:3]: done

File: cypress/e2e/DiagramsPage/focus.movement.diagrams.feature [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    When I click on '[data-cy="component-definitions-item_terrator-plugin"]'
+    And  I click on '[data-cy="component-definition_aws"]'
+    Then I expect '[data-cy="modelizer-draw-view"] [data-cy="draw-container"]' exists
+    And  I expect '[id^="aws"]' exists
+
+    # Back to the diagrams page
+    When I visit the 'localhost:8080/#/{{ projectName }}/diagrams'
+    Then I expect '[data-cy="diagram-card_{{ firstModelName }}"]' exists
+
+    Then I expect '[data-cy="diagram-card_{{ firstModelName }}"]' to be at position 16,16
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/focus.movement.diagrams.feature]: done

**Pull request closed without merging**